server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java

```diff
@@ -134,85 +128,145 @@ protected Void start(Long taskId, TaskService taskService, DelegateExecution exe
         } finally {
             OnlineSchemaChangeContextHolder.clear();
         }
+        updateFlowInstanceStatus(FlowStatus.EXECUTING);
         return null;
     }
 
     @Override
-    protected boolean isSuccessful() {
-        return status == TaskStatus.DONE;
-    }
-
-    @Override
-    protected boolean isFailure() {
-        return status == TaskStatus.FAILED;
-    }
-
-    @Override
-    protected void onFailure(Long taskId, TaskService taskService) {
+    public void onFailure(Long taskId, TaskService taskService) {
         log.warn("Online schema change task failed, taskId={}", taskId);
         super.onFailure(taskId, taskService);
+        updateFlowInstanceStatus(FlowStatus.EXECUTION_FAILED);
     }
 
     @Override
-    protected void onSuccessful(Long taskId, TaskService taskService) {
+    public void onSuccessful(Long taskId, TaskService taskService) {
         log.info("Online schema change task succeed, taskId={}", taskId);
         super.onSuccessful(taskId, taskService);
         updateFlowInstanceStatus(FlowStatus.EXECUTION_SUCCEEDED);
     }
 
     @Override
-    protected void onTimeout(Long taskId, TaskService taskService) {
+    public void onTimeout(Long taskId, TaskService taskService) {
         log.warn("Online schema change task timeout, taskId={}", taskId);
         super.onTimeout(taskId, taskService);
+        updateFlowInstanceStatus(FlowStatus.EXECUTION_EXPIRED);
     }
 
+    /**
+     * refresh task's state and try compete task
+     * 
+     * @param flowInstanceID instance id of flow_instance
+     * @param taskID task id of task_task
+     * @param schedulerID scheduler id of scheduler_scheduler
+     */
+    public void tryCompleteTask(long flowInstanceID, long taskID, long schedulerID) {
+        this.flowInstanceId = flowInstanceID;
+        this.taskId = taskID;
+        this.scheduleId = schedulerID;
+        onProgressUpdate(taskID, taskService);
+        switch (taskStatus) {
+            case ABNORMAL:
+                processAbnormal(taskID, taskService);
+                break;
+            case FAILED:
+                onFailure(taskID, taskService);
```

Comment:
The `onFailure` and `onSuccessful` would be invoked in BaseODCFlowTaskDelegate, And this method is **non** reentrant.
If we call it multiple times, the notification would be resent.

Reply:
Failure and Successful is terminate state, it will only be called once during the life cycle of OSC task.
 it schedule task failed, it will turn into abnormal, no event will be send.